File: Programmars/Level_1/최대공약수와 최소공배수.py
def solution(n, m):
    max_divisor = 0
    min_multiple = 0
    for i in range(1, min(m, n)+1):
        if m % i == 0 and n % i == 0:
            max_divisor = i
    for i in range(m*n, 0, -max(m, n)):
        if i % m == 0 and i % n == 0:
            min_multiple = i
    return [max_divisor, min_multiple]


print(solution(3, 12))  # [3, 12]
print(solution(2, 5))  # [1, 10]


# ref: 유클리드 호제법
def solution(a, b):
    c, d = max(a, b), min(a, b)
    t = 1
    while t > 0:
        t = c % d
        c, d = d, t
    answer = [c, int(a*b/c)]

    return answer

File: JavaScript/Programmars/Level_1/최대공약수와 최소공배수.js
function solution(n, m) {
  const answer = [];
  const [max, min] = [Math.max(n, m), Math.min(n, m)];
  for (let i = min; i >= 1; i--) {
    if (n % i === 0 && m % i === 0) {
      answer.push(i);
      break;
    }
  }
  for (let i = max; i <= n * m; i++) {
    if (i % n === 0 && i % m === 0) {
      answer.push(i);
      break;
    }
  }

  return answer;
}

console.log(solution(3, 12)); // [3, 12]
console.log(solution(2, 5)); // [1, 10]

// ✅ 다른 풀이: 유클리드 호제법
function solution(n, m) {
  let [max, min] = [Math.max(n, m), Math.min(n, m)];
  let t = 1;
  while (t > 0) {
    t = max % min;
    [max, min] = [min, t];
  }

  return [max, Math.floor((n * m) / max)];
}
